platforms/documentation/docs/src/docs/userguide/best-practices/bp_dependencies.adoc

```diff
@@ -15,4 +15,36 @@
 [[best_practices_for_dependencies]]
 = Best Practices for Dependencies
 
-Add content here.
+[[set_up_repositories_in_settings]]
+== Set up your dependency repositories in `settings.gradle.kts`
+
+Declare your repositories for your plugins and dependencies in `settings.gradle.kts`
+
+=== Explanation
+
+Using `settings.gradle.kts` file allows for much clearer structure of what is used within the build and prevents project-level declarations as that often leads to harder to reason behaviors.
```

Comment:
I think this could be expanded to include more detail.  I see the reasons as:

1. DRY coding - the declarations in settings are available to all projects, rather than repeating them in each project.
2. Repos declarations in a single place allows you to locate them easier. (Covered)
3. Repos declarations in a single place allows you to _order_ them so that commonly used repos are first, gaining performance.
4. Defining repos in settings better aligns with the conceptual model - repositories aren't really a part of a project's definition.  They are global build logic, so settings is a more appropriate place for them.

I don't know that behaviors are harder to reason about, depending on where you declare repos - what do you mean by this?  Is it related to 3)?

Reply:
> 3. Repos declarations in a single place allows you to order them so that commonly used repos are first, gaining performance.

I don't quite follow this. You can have identical performance if you declare those repos in build.gradle.kts.

Added the rest of the reasons.

Reply:
For 3) I was thinking if you used mixed declarations and declare repos A, B in settings and repos C, D in your project then I don't think it's well-defined which repo is checked first.

Reply:
> For 3) I was thinking if you used mixed declarations and declare repos A, B in settings and repos C, D in your project then I don't think it's well-defined which repo is checked first.

Given that we don't want people to do that, i'd rather not even discuss that. They should use settings.gradle, or if they really want, each build.gradle, but not both.